### PR TITLE
Support TiFlash Metrics with keyspace info and adjust schema metrics

### LINKED ISSUE #7640: Enhance TiFlashMetrics containing keyspace info
## Enhancement

At present, the metrics can't easily distinguish the value from different keyspace id. It will lead to confusing metrics info in serverless situations. Thus we need to optimize our metrics infrasture, to support metrics with keyspace id.

### PR DESCRIPTION
**Pull request by hongyunyan**
### What problem does this PR solve?

Issue Number: close #7640

Problem Summary:

### What is changed and how it works?
1. expand our metrics infrasutrature, to make it support keyspace info
2. make schema metrics contains keyspace info and adjust the grafana json files.

Now, the grafana for ddl part looks like:
image

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

File: dbms/src/TiDB/Schema/SchemaBuilder.cpp [hunk 1]
```diff
@@ -473,7 +476,7 @@ void SchemaBuilder<Getter, NameMapper>::applyRenamePhysicalTable(
     }
 
     const auto old_mapped_tbl_name = storage->getTableName();
-    GET_METRIC(tiflash_schema_internal_ddl_count, type_rename_column).Increment();
+    GET_KEYSPACE_METRIC(tiflash_schema_internal_ddl_count, type_modify_column, keyspace_id).Increment();
```
Comment (JaySon-Huang) [inline dbms/src/TiDB/Schema/SchemaBuilder.cpp:479]: this should be rename table?

File: metrics/grafana/tiflash_summary.json [hunk 2]
```diff
@@ -5085,33 +5078,33 @@
           "steppedLine": false,
           "targets": [
             {
-              "expr": "avg(increase(tiflash_schema_internal_ddl_count{k8s_cluster=\"$k8s_cluster\", tidb_cluster=\"$tidb_cluster\", instance=~\"$instance\"}[1m])) by (type)",
+              "expr": "avg(increase(tiflash_schema_internal_ddl_count{k8s_cluster=\"$k8s_cluster\", tidb_cluster=\"$tidb_cluster\", instance=~\"$instance\", keyspace!=\"\"}[1m])) by (type, keyspace)",
               "format": "time_series",
               "intervalFactor": 1,
-              "legendFormat": "{{type}}",
+              "legendFormat": "{{type-keyspace}}",
```
Comment (JaySon-Huang) [inline metrics/grafana/tiflash_summary.json:5084]: should it be
```suggestion
              "legendFormat": "{{type}}-{{keyspace}}",
```

File: metrics/grafana/tiflash_summary.json [hunk 3]
```diff
@@ -4880,10 +4880,10 @@
           "steppedLine": false,
           "targets": [
             {
-              "expr": "avg(tiflash_schema_version{k8s_cluster=\"$k8s_cluster\", tidb_cluster=\"$tidb_cluster\", instance=~\"$instance\"}) by (instance)",
+              "expr": "avg(tiflash_schema_version{k8s_cluster=\"$k8s_cluster\", tidb_cluster=\"$tidb_cluster\", instance=~\"$instance\", keyspace!=\"\"}) by (instance, keyspace)",
               "format": "time_series",
               "intervalFactor": 1,
-              "legendFormat": "{{instance}}",
+              "legendFormat": "{{instance-keyspace}}",
```
Comment (JaySon-Huang) [inline metrics/grafana/tiflash_summary.json:4886]: ditto

**Review by lidezhu — APPROVED** [review]
LGTM

**Review by JaySon-Huang — APPROVED** [review]

**Comment by SeaRise** [comment]
It looks like the metrics for query level should also include keyspace info.

**Pull request merged**